statsmodels/regression/tests/test_quantile_regression.py

```diff
@@ -281,3 +281,15 @@ def test_alpha_summary():
     summ_20 = res.summary(alpha=.2)
     assert '[0.025      0.975]' not in str(summ_20)
     assert '[0.1        0.9]' in str(summ_20)
+
+
+def test_collinear_matrix():
+    X = np.array([[1, 0, .5], [1, 0, .8],
+                  [1, 0, 1.5], [1, 0, .25]], dtype=np.float64)
+    y = np.array([0, 1, 2, 3], dtype=np.float64)
+
+    try:
+        est_collinear = QuantReg(y, X).fit(0.5)
+    except Exception as e:
```

Comment:
what exception does this raise in the singular case?

using rank  before the fix was an obvious bug, 
However, it is not clear how the singular exog case works or should work

Reply:
yeah that's not 100% clear to me yet (this PR was in part aimed to check whether unit tests passed with the change, and start the conversation...still a work in progress)

when I run with this change with a singular design matrix, the only obvious thing that comes up is that line 199 (` h = bandwidth(nobs, q)`) comes up as nan with the default bandwidth, which raises a RuntimeWarning but no Exception. 

It seems like at minimum an appropriate warning or error should be raised so the user knows that they passed a singular design matrix to the model, but I'm also not certain how much IRLS is impacted by a singular matrix

Reply:
> ( h = bandwidth(nobs, q)) comes up as nan

this sounds strange, this bandwidth does not depend on the data, just on nobs

Reply:
Ah! I misspoke - it's not bandwidth itself that comes up nan, but the call to `norm.ppf(q+h)` on line 200.
but changing the bandwidth does fix the problem....and I think it's because for some reason that default bandwidth is resulting in an h val that's greater than q, which leads us to take the pdf of numbers that are >1 and <0

Reply:
Good, I guess this is then an issue with the small sample size that you use. There might be a minimum sample size that the bandwidth creates a valid value.
You could open a separate issue for that if I guess correctly

Reply:
...yep, based on some quick-and-dirty testing it looks like this can have issues with relatively low sample size. I may open an issue in lieu of doing more rigorous testing to find that boundary.